.github/scripts/run-parallel-vcs-metasims.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python3
+
+import sys
+from pathlib import Path
+
+from fabric.api import prefix, run, settings, execute # type: ignore
+
+from ci_variables import ci_env
+
+def run_parallel_metasim():
+    """ Runs parallel baremetal metasimulations """
+
+    # assumptions:
+    #   - machine-launch-script requirements are already installed
+    #   - VCS is installed
+
+    # repo should already be checked out
+
+    with prefix(f"cd {ci_env['GITHUB_WORKSPACE']}"):
+        run("./build-setup.sh --skip-validate")
+        with prefix('source sourceme-f1-manager.sh --skip-ssh-setup'):
+            # avoid logging excessive amounts to prevent GH-A masking secrets (which slows down log output)
+            with prefix('cd sw/firesim-software'):
+                run("./init-submodules.sh")
+
+                # build hello world baremetal test
+                with settings(warn_only=True):
+                    rc = run("./marshal -v build test/bare.yaml &> bare.full.log").return_code
+                    if rc != 0:
+                        run("cat bare.full.log")
+                        raise Exception("Building test/bare.yaml failed to run")
+
+                run("./marshal -v install test/bare.yaml")
+
+            run("firesim managerinit --platform vitis")
+
+            def run_w_timeout(workload: str, timeout: str):
```

Comment:
This is largely the same as the `run_w_timeout` in the fpga-sim CI scripts right? I have a PR in our fork right now that helps pull some of this out, but would be helpful trying to understand any differences between them.

Reply:
For the most part yes. The main difference is that the actual "run workload" line (https://github.com/firesim/firesim/blob/ae9ea901880fcf7762c69da32c2efae40eff0730/.github/scripts/run-linux-poweroff-vitis.py#L44) in the FPGA sim case is setup to point to a different `run-workload.sh` script with a different set of files (see the function args) compared to this function which mainly wraps around the existing `deploy/workloads/run-workload.sh` script.